sources/api/schnauzer/src/helpers/mod.rs

```diff
@@ -535,6 +542,77 @@ pub fn tuf_prefix(
     Ok(())
 }
 
+/// The `aws_config` helper is used to create an AWS config file
+/// with `use_fips_endpoint` value set based on if the variant is FIPS enabled.
+///
+/// # Fallback
+///
+/// If this helper runs and `settings.aws.config` is set,
+/// the helper will return early, leaving the existing content intact.
+///
+/// # Example
+///
+/// The AWS config value is generated via
+/// `{{ aws_config settings.aws.config settings.aws.profile }}`
+///
+/// This would result in something like:
+/// ```toml
+/// [default]
+/// use_fips_endpoint=false
+/// ```
+///
+/// The helper will then base64 encode this content and set as `settings.aws.config`.
+pub fn aws_config(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    trace!("Starting aws-config-fips-endpoint helper");
+    let template_name = template_name(renderctx);
+
+    check_param_count(helper, template_name, 2)?;
+
+    let aws_config = get_param(helper, 0)?;
+    if let Value::Null = aws_config {
```

Comment:
nit: I'd recommend an early return here if `aws_config` is not null, to avoid rightward drift.

Reply:
Sure, I'll instead make this a `match` block and return early on `Value::Null`